chain/solana/src/runtime/generated.rs

```diff
@@ -0,0 +1,306 @@
+    pub last_entry_hash: AscPtr<Uint8Array>,
+    pub transactions: AscPtr<AscTransactionArray>,
+    pub account_changes_file_ref: AscPtr<AscString>,
+    pub has_split_account_changes: bool,
+    pub _padding: Padding3,
+}
+
+impl AscIndexId for AscBlock {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::SolanaBlock;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscInstruction {
+    pub ordinal: u32,
+    pub parent_ordinal: u32,
+    pub depth: u32,
+    pub program_id: AscPtr<AscHash>,
+    pub account_keys: AscPtr<AscHashArray>,
+    pub data: AscPtr<AscHash>,
+    pub balance_changes: AscPtr<AscBalanceChangeArray>,
+    pub account_changes: AscPtr<AscAccountChangeArray>,
+    pub error: AscPtr<AscInstructionError>,
+    pub failed: bool,
+    pub _padding: Padding3,
+}
+
+// special type for the rare occurance where we need to pad three bytes
+pub(crate) struct Padding3([u8; 3]);
+
+impl Padding3 {
+    pub fn new() -> Self {
+        Padding3 { 0: [0, 0, 0] }
+    }
+}
+
+impl AscType for Padding3 {
+    fn to_asc_bytes(&self) -> Result<Vec<u8>, DeterministicHostError> {
+        Ok(vec![0, 0, 0])
+    }
+
+    fn from_asc_bytes(
+        _asc_obj: &[u8],
+        _api_version: &Version,
+    ) -> Result<Self, DeterministicHostError> {
+        Ok(Padding3 { 0: [0, 0, 0] })
+    }
+}
+
+// special type for the rare occurance where we need to pad three bytes
+pub(crate) struct Padding7([u8; 7]);
+
+impl Padding7 {
+    pub fn new() -> Self {
+        Padding7 {
+            0: [0, 0, 0, 0, 0, 0, 0],
+        }
+    }
+}
+
+impl AscType for Padding7 {
+    fn to_asc_bytes(&self) -> Result<Vec<u8>, DeterministicHostError> {
+        Ok(vec![0, 0, 0, 0, 0, 0, 0])
+    }
+
+    fn from_asc_bytes(
+        _asc_obj: &[u8],
+        _api_version: &Version,
+    ) -> Result<Self, DeterministicHostError> {
+        Ok(Padding7 {
+            0: [0, 0, 0, 0, 0, 0, 0],
+        })
+    }
+}
+
+impl AscIndexId for AscInstruction {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::SolanaInstruction;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscInstructionWithInfo {
+    pub block_num: u64,
+    pub instruction: AscPtr<AscInstruction>,
+    pub block_id: AscPtr<Uint8Array>,
+    pub transaction_id: AscPtr<Uint8Array>,
+    pub _padding: u32,
+}
+
+impl AscIndexId for AscInstructionWithInfo {
+    const INDEX_ASC_TYPE_ID: IndexForAscTypeId = IndexForAscTypeId::SolanaInstruction;
+}
+
+#[repr(C)]
+#[derive(AscType)]
+pub(crate) struct AscTransaction {
+    pub index: u64,
+    pub id: AscPtr<Uint8Array>,
+    pub additional_signatures: AscPtr<AscHashArray>,
+    pub header: AscPtr<AscMessageHeader>,
+    pub account_keys: AscPtr<AscHashArray>,
+    pub recent_blockhash: AscPtr<Uint8Array>,
+    pub log_messages: AscPtr<AscStringArray>,
+    pub instructions: AscPtr<AscInstructionArray>,
+    pub error: AscPtr<AscTransactionError>,
+    pub failed: bool,
```

Comment:
Double check ordering of fields in `graph-ts`, in the version of the branch I have on `graph-ts`, those are ordered differently as far as I remember (`error` is after `failed`)

Reply:
ordering is the same on the latest commit in graph-ts